[docs] k8s pod eviction (#32767)
dagster-io/dagster: docs/docs/deployment/oss/deployment-options/kubernetes/managing-pod-eviction.md

---
title: 'Handling pod eviction'
description: How to handle pod eviction in Kubernetes deployments.
sidebar_position: 4400
---

> When running Dagster in Kubernetes environments, pod evictions can occur due to resource constraints or node scaling operations. To prevent runs from hanging indefinitely, you can implement the following solutions:

Reviewer: ```suggestion
When Dagster is run in Kubernetes environments, pod evictions can occur due to resource constraints or node scaling operations. To prevent runs from hanging indefinitely, you can implement the following solutions:
```

```
"cluster-autoscaler.kubernetes.io/safe-to-evict": "false"

> ```

Reviewer: ```suggestion

```

> 

Reviewer: ```suggestion

For more information, see the [Kubernetes docs](https://kubernetes.io/docs/reference/labels-annotations-taints/#cluster-autoscaler-kubernetes-io-safe-to-evict).

```

Configure job-specific timeouts in your Dagster configuration to ensure runs don't hang indefinitely:

```python
@job(

>     config={

Reviewer: ```suggestion

```

>         "execution": {

Reviewer: ```suggestion

```

>             "config": {

Reviewer: ```suggestion

```

                "max_concurrent": 1,
                "timeout_seconds": 7200  # 2 hours

>             }

Reviewer: ```suggestion

```

>         }

Reviewer: ```suggestion

```

>     }

Reviewer: ```suggestion

```

)

> ```

Reviewer: ```suggestion

```

> Pod evictions can occur due to:

Reviewer: ```suggestion

```

- Node running out of memory

> - Node scale-down operations

Reviewer: ```suggestion

```
